src/Functions/keyvaluepair/src/KeyValuePairExtractor.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <string>
+#include <unordered_map>
+
+namespace DB
+{
+/*
+ * Extracts key value pairs from strings. Strings do not need to be key-value pair only,
+ * it can contain "noise". The below grammar is a simplified representation of what is expected/ supported:
+ *
+ * line = (reserved_char* key_value_pair)*  reserved_char*
+ * key_value_pair = key kv_separator value
+ * key = <quoted_string> |  asciichar asciialphanumeric*
+ * kv_separator = ':'
+ * value = <quoted_string> | asciialphanum*
+ * item_delimiter = ','
+ *
+ * Both key and values accepts underscores as well. Special characters must be escaped.
+ * Control characters (key_value_pair_separator, item_delimiter, escape_character and enclosing_character) are customizable
+ * */
+template <typename R = std::unordered_map<std::string, std::string>>
+struct KeyValuePairExtractor
```

Comment:
Seems that `LazyEscapingKeyValuePairExtractor` is only inheritor, can't we not use this interface at all and expose just function `std::unordered_map<std::string_view, std::string_view> exctractKeyValuePairs(const std::string & file, KeyValueSettings settings)` ?